src/app/components/mainpage/mainpage.component.html

```diff
@@ -1,89 +1,126 @@
-            </button>
-          </li>
-          <li *ngIf="isLoggedIn" class="nav-item m-2">
-            <button (click)="onLogOut()" class="btn btn-sm btn-success text-white m-3 p-3 ">Logout
-            </button>
-          </li>
-          <li *ngIf="isLoggedIn" class="nav-item m-2">
-            <button (click)="displayAccountInfo()" class="btn btn-sm btn-success text-white m-3 p-3 ">Account Info</button>
-          </li>
-          <li *ngIf="isLoggedIn && isAdmin" class="nav-item m-2">
-            <button (click)="gotoAdminConsole()"
-                    class="btn btn-sm btn-success text-white m-3 p-3 ">Admin
-            </button>
-          </li>
-        </ul>
-      </div>
+<nav class="navbar navbar-expand-sm navbar-dark bg-primary pt-4">
+  <a href="/" class="navbar-brand text-white d-none d-md-block">Interview Question Generator</a>
+  <a href="/" class="navbar-brand text-white d-md-none">IQG</a>
+  <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
+    <span class="navbar-toggler-icon"></span>
+  </button>
+  <div class="collapse navbar-collapse" id="navbarSupportedContent">
+    <ul *ngIf="!isLoggedIn" class="navbar-nav ml-auto">
+      <li class="nav-item">
+        <a class="nav-link" href="#" (click)="displayLogin(); false">Login</a>
+      </li>
+      <li class="nav-item">
+        <a class="nav-link" href="#" (click)="displayRegistration(); false">Register</a>
+      </li>
+    </ul>
+    <ul *ngIf="isLoggedIn" class="navbar-nav ml-auto">
+      <li class="nav-item">
+        <a class="nav-link" href="#" (click)="onLogOut(); false">Logout</a>
+      </li>
+      <li class="nav-item">
+        <a class="nav-link" href="#" (click)="displayAccountInfo(); false">Account Info</a>
+      </li>
+      <li *ngIf="isAdmin" class="nav-item">
+        <a class="nav-link" href="#" (click)="gotoAdminConsole(); false">Admin</a>
+      </li>
+    </ul>
+  </div>
+</nav>
+<div class="container">
+  <div class="row py-2">
+    <div class="col d-none d-lg-block my-2">
+      <h1>Welcome to the Interview Question Generator!</h1>
     </div>
-  </nav>
-  <div class="container-fluid">
-    <div class="row">
-      <div class="col col-sm-4">
-        <div class="card bg-white border-white">
-          <h2 class="text-left">Select a Category and Difficulty Level</h2>
+  </div>
+  <div class="row">
+    <div class="col-lg-4 mb-2">
+      <form (submit)="onSubmit($event)" #Filter="ngForm">
+        <div class="form-group d-lg-none">
+          <div class="container-fluid">
+            <div class="row">
+              <div class="col">
+                <select class="form-control custom-select"
+                    [(ngModel)]="filterCategoryValue"
+                    required
+                    name="questionCategoryId_dropdown">
+                    <option disabled selected>Category</option>
```

Comment:
I have no idea why these options for Category and Difficulty aren't showing selected on page load